packages/vendure-plugin-google-cloud-tasks/src/cloud-tasks-job-queue.strategy.ts

```diff
@@ -1,21 +1,125 @@
 import { CloudTasksClient } from '@google-cloud/tasks';
-import { Job, JobData, JobQueueStrategy, Logger } from '@vendure/core';
+import {
+  ID,
+  Injector,
+  InspectableJobQueueStrategy,
+  Job,
+  JobData,
+  JobQueueStrategy,
+  ListQueryBuilder,
+  Logger,
+  PaginatedList,
+  TransactionalConnection,
+  UserInputError,
+} from '@vendure/core';
 import { CloudTasksPlugin } from './cloud-tasks.plugin';
 import { CloudTaskMessage, CloudTaskOptions } from './types';
-import { JobState } from '@vendure/common/lib/generated-types';
+import { In, LessThan } from 'typeorm';
+import { JobListOptions, JobState } from '@vendure/common/lib/generated-types';
+import { JobRecord } from '@vendure/core/dist/plugin/default-job-queue-plugin/job-record.entity';
 
 const LIVE_QUEUES = new Set<string>();
 
 export type QueueProcessFunction = (job: Job) => Promise<any>;
 export const PROCESS_MAP = new Map<string, QueueProcessFunction>();
 
-export class CloudTasksJobQueueStrategy implements JobQueueStrategy {
+export class CloudTasksJobQueueStrategy implements InspectableJobQueueStrategy {
   private client: CloudTasksClient;
+  private connection: TransactionalConnection | undefined;
+  private listQueryBuilder: ListQueryBuilder | undefined;
+
+  init(injector: Injector): void | Promise<void> {
+    this.connection = injector.get(TransactionalConnection);
+    this.listQueryBuilder = injector.get(ListQueryBuilder);
+  }
 
   constructor(private options: CloudTaskOptions) {
     this.client = new CloudTasksClient();
   }
 
+  async findOne(id: ID): Promise<Job<any> | undefined> {
+    if (!this.connection) {
+      throw new UserInputError('TransactionalConnection is not available');
+    }
+    const jobRecord = await this.connection
+      .getRepository(JobRecord)
+      .findOne({ where: { id } });
```

Comment:
These are deprecated. because we dont have a `ctx` available, we can use this:
```suggestion
    const jobRecord = await this.connection.rawConnection
      .getRepository(JobRecord)
      .findOne({ where: { id } });
```